### PR TITLE
Potential fix/workaround for the "connection remotely closed" netty bug.

### PR DESCRIPTION
**Pull request by stkem**
Fingers crossed. @gmethvin, @ymatsuda please have a look.

### CODE REVIEW

**Comment by 2is10** [comment]
A comment explaining how the future/async avoids the bug and containing a link to the bug would be helpful.

**Comment by stkem** [comment]
This is really just a hail marry. There are several reports (some filed as bugs with netty) on this issue but no one knows for sure what the problem is. The most common theory is that some socket or file handle does not get released properly and when netty tries to use it again things go wrong.
On our end the problem only seems to occur when calling "/site/keeps/all", so I tried to figure out what was special about that call. The only thing unusual I found was that it makes a call to search which could potentially return a lot of data (all friends who keep one of your keeps, I think) and that was handled synchronously. It's somewhere in this call that the problem occurs.
The Play documentation highly discourages doing anything that does any kind of significant computation in a synchronous way (I think it blocks a netty thread), so I changed the call and now I'm hoping for the best :-)

**Comment by gmethvin** [comment]
Okay, I suppose this is worth a try.

File: kifi-backend/modules/search/app/com/keepit/controllers/search/URIGraphController.scala [hunk 1]
```diff
@@ -43,9 +43,14 @@ class URIGraphController @Inject()(
   }
 
   def sharingUserInfo(userId: Id[User], uriIds: String) = Action { implicit request =>
-    val searcher = mainSearcherFactory.getURIGraphSearcher(userId)
-    val ids = uriIds.split(",").map(_.trim).collect { case idStr if !idStr.isEmpty => Id[NormalizedURI](idStr.toLong) }
-    Ok(Json.toJson(ids map searcher.getSharingUserInfo))
+    val jsonFuture = future {
```
Comment (gmethvin) [inline kifi-backend/modules/search/app/com/keepit/controllers/search/URIGraphController.scala:46]: I would just wrap the whole thing in `Async` and `future`.

``` scala
Async(future {
  val searcher = mainSearcherFactory.getURIGraphSearcher(userId)
  val sharingUserInfos = uriIds.split(",").map(_.trim).filterNot(_.isEmpty).map { idStr =>
    searcher.getSharingUserInfo(Id[NormalizedURI](idStr.toLong))
  }
  Ok(Json.toJson(sharingUserInfos))
})
```

Comment (stkem) [inline kifi-backend/modules/search/app/com/keepit/controllers/search/URIGraphController.scala:46]: I actually like this better in two steps. Reduced nesting level, easier to read, imho.

Comment (gmethvin) [inline kifi-backend/modules/search/app/com/keepit/controllers/search/URIGraphController.scala:46]: Sure; was just a suggestion. It's personal preference I guess.

**Comment by gmethvin** [comment]
lgtm

**Pull request merged**